### PR TITLE
Feature/1200973179398805/signup-flow

### PR DESCRIPTION
**Pull request by trigal2012**
## Description

sign up flow redesign - updated the name of the PR to indicate that this is for all of the changes, not just one component

## Asana ticket:
https://app.asana.com/0/1132189118126148/1169561089123233

## PR Checklist

- [x] Assign @FJKhan **and** @Alfredo-Moreira as reviewers.
- [x] If your PR is not a hotfix, is it targeted for `dev`? If it is a hotfix, is it targeted for `main`?
- [x] Unit and functional test coverage was added where applicable.
- [x] CI/CD passes for your PR.
- [x] Complex code is well documented with comments.
- [x] Does the original ticket have test instructions? If not add them below
- [x] Pass QA Gate(manual testing)

## How to Test

following the design flow here:
https://www.figma.com/file/WvHG6rIzsOibNUgl8rBFC1/%5BDONE---Mar-'20%5D-Signup-flow?node-id=0%3A1

**Changes to note:**
The following are single select Radio buttons:
- Sign up screen - lawyer and provider - Where do you practice law/volunteer
- About You - Immigration Status, Country of Origin, Age

The following are multi-select Checkboxes:
- About You - Identity, Ethnicity

**General Rules:**
- if there is a "Prefer not to say" option and the available options are multi-select - when "Prefer not to say" is checked, all other options are disabled
- If there is an "Other" option - if "Other" is selected a text field will appear. This text field must contain at least 2 characters in order for the "next/submit" button to be enabled. Even though none of the "About You" answers are required, the thinking here is that if a user goes thru the trouble of selecting "Other" then they should provide a value in the text field.

This flow has a few phases, depending on the seeker type:
- For Myself - Sign up -> About You -> Thank You
- For Lawyer/Provider - Sign up -> Select Organization -> About You -> Thank You
- The back button allows the user to go to a previous view within a phase, but once they complete a phase, they can't go back. For example, Lawyer/Provider has the initial view, plus two signup modals. Once they click submit on the 2nd Sign Up modal, they then enter the Select Organization phase. Once in the Organization phase there is not a back button that goes back to the sign up modal.

A new user is created in the DB at the end of the Signup phase (handleSignup())
That user is then updated at each step of the About You phase (handleUpdateUser())

For the multi-select options, data is saved in the DB as an array of strings.
Here is an example of the About You - Identity with 3 of the predefined options selected: 
Screen Shot 2022-01-06 at 7 09 28 PM

Here is an example of the About You - Identity - with Nonbinary and Other selected:
Screen Shot 2022-01-06 at 7 11 14 PM

**Checkbox and Radio button options**
The data for these options are kept in some data files located in the data dir:
../asylumconnect-catalog/src/data/aboutYouFormOptions.js
../asylumconnect-catalog/src/data/organizationTypeFormOptions.js

data is an array of objects: (i.e)
```
export const organizationTypesLawyer = [
	{
		dbValue: 'corporate-law',
		formatMessageId: 'account.sign-up-orgType-answer-corp'
	},
	{
		dbValue: 'law-school-clinic',
		formatMessageId: 'account.sign-up-orgType-answer-law'
	},
	{
		dbValue: 'legal-non-profit',
		formatMessageId: 'account.sign-up-orgType-answer-legal-non-profit'
	},
	{
		dbValue: 'other',
		formatMessageId: 'aboutyou.answer-other'
	}
];
```

The 'dbValue' is the value saved to the DB.

### CODE REVIEW

File: src/components/NameEmailPswd.js [hunk 1]
```diff
@@ -0,0 +1,332 @@
...
+	const [touchedEmail, setTouchedEmail] = useState(false);
+	const [touchedPassword, setTouchedPassword] = useState(false);
+
+	const textFieldTest = new RegExp(/\s*(?:[\S]\s*){2}$/);
+	const emailTest = new RegExp(/\S+@\S+\.\S+/);
+	const pswdTest = new RegExp(
+		'^(?=.*[a-z])(?=.*[A-Z])(?=.*[0-9])(?=.*[!@#$%^&?])(?=.{10,})'
+	);
+
+	const handleTouchName = () => {
```
Comment (FJKhan) [inline src/components/NameEmailPswd.js]: these handlers aren't necessary. it'd be better to use the hook directly

File: src/components/NameEmailPswd.js [hunk 2]
```diff
@@ -0,0 +1,332 @@
...
+					}}
+				/>
+				<FormLabel required className={classes.labels} margin="none">
+					<FormattedMessage id="form.password" />
+				</FormLabel>
+				<TextField
+					onBlur={handleTouchPassword}
+					error={touchedPassword && pswdTest.test(password) === false}
+					helperText={
+						touchedPassword && pswdTest.test(password) === false ? (
```
Comment (FJKhan) [inline src/components/NameEmailPswd.js]:
```suggestion
						touchedPassword && !pswdTest.test(password)  ? (
```

File: src/components/NameEmailPswd.js [hunk 3]
```diff
@@ -0,0 +1,332 @@
...
+							<FormattedMessage id="error.password-format" />
+						) : touchedPassword && pswdTest.test(password) === true ? (
```
Comment (FJKhan) [inline src/components/NameEmailPswd.js]:
```suggestion
						) : touchedPassword && pswdTest.test(password) ? (
```

File: src/components/NameEmailPswd.js [hunk 4]
```diff
@@ -0,0 +1,332 @@
...
+					InputProps={{
+						classes: {
+							input: classes.borderOutline,
+							notchedOutline: classes.borderOutline
+						}
+					}}
+				/>
+				<AsylumConnectSignupAgreement />
+				<AsylumConnectButton
+					disabled={isValid() === false ? true : false}
```
Comment (FJKhan) [inline src/components/NameEmailPswd.js]:
```suggestion
					disabled={!isValid()}
```

File: src/components/NameLocationLawyerProvider.js [hunk 5]
```diff
@@ -0,0 +1,390 @@
...
+			? organizationTypesLawyer
+			: organizationTypesProvider;
+	const orgTypeOther =
+		selection === LAWYER_TYPE ? (
+			<FormattedMessage id="account.signup-organization-orgType-lawyer-other" />
+		) : (
+			<FormattedMessage id="account.signup-organization-orgType-provider-other" />
+		);
+
+	const handleTouchName = () => {
```
Comment (FJKhan) [inline src/components/NameLocationLawyerProvider.js]: these handlers aren't necessary. use the hook directly

File: src/components/OrganizationAutocomplete.js [hunk 6]
```diff
@@ -18,8 +20,17 @@ function renderInput(inputProps) {
 	const {classes, value, ref} = inputProps;
 	return (
 		<FormControl className={classes.textField}>
-			<InputLabel htmlFor="organization" className={classes.textFieldLabel}>
-				Organization Name
+			<InputLabel
+				htmlFor="organization"
+				className={
```
Comment (FJKhan) [inline src/components/OrganizationAutocomplete.js]: use interpolated string

```suggestion
				className={ `${classes.textFieldLabel} ${classes.textAlignLeft} ${classes.colorGrey}` }
```

File: src/components/SignupFormContainer.js [hunk 7]
```diff
@@ -69,34 +211,142 @@ class SignupFormContainer extends React.Component {
...
+
 		const isProfessional = selection === 'lawyer' || selection === 'provider';
+		const emailTest = new RegExp(/\S+@\S+\.\S+/);
+		const pswdTest = new RegExp(
+			'^(?=.*[a-z])(?=.*[A-Z])(?=.*[0-9])(?=.*[!@#$%^&?])(?=.{10,})'
+		);
 
-		if (password.length < 8) {
-			handleMessageNew(<FormattedMessage id="error.password-length" />);
+		if (pswdTest.test(password) === false) {
```
Comment (FJKhan) [inline src/components/SignupFormContainer.js]:
```suggestion
		if (!pswdTest.test(password)) {
```

File: src/components/SignupFormContainer.js [hunk 8]
```diff
@@ -69,34 +211,142 @@ class SignupFormContainer extends React.Component {
...
+			handleMessageNew(<FormattedMessage id="error.password-format" />);
 			return;
 		}
 
-		if (password !== passwordConfirmation) {
-			handleMessageNew(<FormattedMessage id="error.password-mismatch" />);
+		if (emailTest.test(email) === false) {
```
Comment (FJKhan) [inline src/components/SignupFormContainer.js]: `test()` returns a boolean. you don't need to compare it, just use the returned value to make assertions.
```suggestion
		if (!emailTest.test(email)) {
```

File: src/data/aboutYouFormOptions.js [hunk 9]
```diff
@@ -0,0 +1,205 @@
...
+	{
+		testId: 'other',
+		formatMessageId: 'aboutyou.answer-other'
+	}
+];
+
+export const aboutYouEthnicityOptions = [
+	{
+		testId: 'af-md',
+		formatMessageId: 'aboutyou.ehnicity-answer-af-md'
```
Comment (FJKhan) [inline src/data/aboutYouFormOptions.js]: "ethnicity" is misspelt in all options

```suggestion
		formatMessageId: 'aboutyou.ethnicity-answer-af-md'
```

**Review by FJKhan — DISMISSED** [review]

**Comment by FJKhan** [comment]
re: data storage -- ideally, we need the plain text responses as the whole point of capturing this data is to generate metrics. while `aboutyou.identity-anser-non-binary` is useful for localization, the data would be harder to parse for those performing analysis.

i.e. instead of `aboutyou.identity-anwser-non-binary`, we'd store "non-binary"

**Review by Alfredo-Moreira — APPROVED** [review]
Looks Good to me. Nice work!

**Pull request merged**